main.tf

```diff
@@ -9,8 +12,16 @@ locals {
     id_hash_length      = 5
     label_key_case      = "title"
     label_value_case    = "lower"
+
+    # The default value of labels_as_tags cannot be included in this
+    # defaults` map because it creates a circular dependency
   }
 
+  default_labels_as_tags = keys(local.tags_context)
+  # Unlike other inputs, the first setting of `labels_as_tags` cannot be later overridden
+  # We have to cover 2 cases. 1) context does not have a labels_as_tags key, 2) it is present and set to ["default"]
```

Comment:
a little confused about "default" and "unset" in `labels_as_tags`

Reply:
Hopefully clearer now.